blueprints/data-solutions/dataproc-lifecycle-via-composer/main.tf

```diff
@@ -0,0 +1,92 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+locals {
+  _shared_vpc_bindings = {
```

Comment:
Reading documentation, those are roles needed on the Service Project (where you deploy composer).

For a deployment on a shared VPC, you need to bing roles on the Service Identities service account. Something similar to this should work:
```
  iam = { # Assigned on the Service Project
  ...
      "roles/composer.worker"            = [module.comp-sa.iam_email]
      "roles/composer.ServiceAgentV2Ext" = ["serviceAccount:${module.project.service_accounts.robots.composer}"]
    }

  _shared_vpc_bindings = { # Assigned on the Host project
    "roles/compute.networkUser" = [
      "prj-cloudservices", "prj-robot-gke", "prj-dataproc"
    ]
    "roles/composer.sharedVpcAgent" = [
      "prj-robot-cs"
    ]
    "roles/container.hostServiceAgentUser" = [
      "prj-robot-gke"
    ]
  }
```